### PR TITLE
[PTQ][MinMax][Torch] One shared quantizer is used for all unified scale quantization points

### PR DESCRIPTION
**Pull request by daniil-lyakhov**
### Changes

* MinMax: new backend method `create_unified_scales_quantizers_insertion_commands` is introduced: it receives several target points and one quantization parameter. Depending on implementation, one or several insertion commands are generated and returned back to the common algorithm. 

### Reason for changes

* Torch backend requires one `PTSharedFNInsertionCommand`  to make quantizers aligned during QAT in comparison with OV/ONNX backend, which can use separate commands/quantizers for each insertion point without any restrictions

### Related tickets

104304

### Tests

[Template test] test_ptq_params: test_unified_scales_command_creation
test_create_shared_quantizer_insertion_command

### Jobs
manual/job/post_training_quantization/350/: passed

### CODE REVIEW

File: nncf/quantization/algorithms/min_max/backend.py [hunk 1]
```diff
@@ -141,7 +141,25 @@ def create_quantizer_insertion_command(
...
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_unified_scales_quantizers_insertion_commands(
+        nncf_graph: NNCFGraph,
+        target_points: List[TargetPoint],
+        quantizer_config: QuantizerConfig,
+        parameters: FakeQuantizeParameters,
+    ) -> TransformationCommand:
```
Comment (KodiaqQ) [inline nncf/quantization/algorithms/min_max/backend.py:157]:
```suggestion
    ) -> List[TransformationCommand]:
```

Comment (daniil-lyakhov) [reply nncf/quantization/algorithms/min_max/backend.py]: Thanks! Done

**Review by KodiaqQ — APPROVED** [review]

File: tests/post_training/test_templates/test_ptq_params.py [hunk 2]
```diff
@@ -91,6 +93,39 @@ def __init__(self, metatypes: Dict[TestMetatype, OperatorMetatype]):
...
+class ModelWithUnifiedScales:
+    #   Input_1 -----------
+    #      |     \         \
+    #   Conv_1   Conv_2  Conv_3
+    #        \         / |
+    #          \      /  |
+    #              \ /   /
+    #              Cat_1
+    #                |
+    #           Output_1
```
Comment (AlexanderDokuchaev) [inline tests/post_training/test_templates/test_ptq_params.py:105]:
```suggestion
    #        Input_1
    #       /   |   \
    #  Conv_1 Conv_2 Conv_3
    #       \   |   /
    #         Cat_1
    #           |
    #        Output_1
```

File: tests/post_training/test_templates/test_ptq_params.py [hunk 3]
```diff
@@ -335,3 +413,14 @@ def get_statistics(self):
             algo.apply(None, None, stat_points)
 
         assert str(exc_info.value) == "Statistics were not collected for the node A"
+
+    class DummyMinMaxTensorStatistic(MinMaxTensorStatistic):
```
Comment (AlexanderDokuchaev) [inline tests/post_training/test_templates/test_ptq_params.py:417]: Please define dummy classes outside of template class or before test functions

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

**Pull request merged**